addons/web/tests/test_db_manager.py

```diff
@@ -110,10 +116,72 @@ def test_database_duplicate(self):
         self.assertDbs([test_db_name])
 
         # delete the created database
-        res = self.session.post(self.url('/web/database/drop'), data={
-            'master_pwd': self.password,
-            'name': test_db_name,
-        }, allow_redirects=False)
-        self.assertEqual(res.status_code, 303)
+        res = self.url_open_drop(test_db_name)
         self.assertIn('/web/database/manager', res.headers['Location'])
         self.assertDbs([])
+
+    def test_database_restore(self):
+        test_db_name = self.db_name + '-test-database-restore'
+        self.assertNotIn(test_db_name, self.list_dbs_filtered())
+
+        # backup the current database inside a temporary zip file
+        res = self.session.post(
+            self.url('/web/database/backup'),
+            data={
+                'master_pwd': self.password,
+                'name': self.db_name,
+            },
+            allow_redirects=False,
+            stream=True,
+        )
+        res.raise_for_status()
+        datetime_pattern = r'\d\d\d\d-\d\d-\d\d_\d\d-\d\d-\d\d'
+        self.assertRegex(
+            res.headers.get('Content-Disposition'),
+            fr"attachment; filename\*=UTF-8''{self.db_name}_{datetime_pattern}\.zip"
+        )
+        backup_file = tempfile.TemporaryFile(prefix=self.db_name, suffix='.zip')
```

Comment:
I think that this could work using io.StringIO instead, we don't need a file just a file descriptor if i'm correct. 

This will avoid having a red ci/security + potential disc access

Reply:
well as we are exporting the current database there is no guarantee that it'll fit in memory but I don't expect people in production will run this test, BytesIO is fine